cookbooks/agents_with_memory.ipynb

```diff
@@ -0,0 +1,728 @@
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "### Adding `LongtermAgentMemory` to your `ChatAgent`:"
+      ],
+      "metadata": {
+        "id": "K-P9rD6Foe5j"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "from camel.agents import ChatAgent\n",
+        "\n",
+        "# Define system message for the agent\n",
+        "sys_msg = BaseMessage.make_assistant_message(\n",
+        "    role_name='Agent',\n",
+        "    content='You are a curious agent wondering about the universe.',\n",
+        ")\n",
+        "\n",
+        "# Initialize agent\n",
+        "agent = ChatAgent(system_message=sys_msg)\n",
+        "\n",
+        "# Set memory to the agent\n",
+        "agent.memory = memory\n",
+        "\n",
+        "\n",
+        "# Define a user message\n",
+        "usr_msg = BaseMessage.make_user_message(\n",
+        "    role_name='User',\n",
+        "    content=\"Tell me which is the 1st LLM multi-agent framework based on what we have discussed\",\n",
+        ")\n",
+        "\n",
+        "# Sending the message to the agent\n",
+        "response = agent.step(usr_msg)\n",
+        "\n",
+        "# Check the response (just for illustrative purpose)\n",
+        "print(response.msgs[0].content)"
+      ],
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "id": "Mp-KXFdGpKZt",
+        "outputId": "e0e6c8b2-142e-4a3a-d34d-2feafebac137"
+      },
+      "execution_count": 5,
+      "outputs": [
+        {
+          "output_type": "stream",
+          "name": "stdout",
+          "text": [
+            "CAMEL AI is recognized as the first LLM (Large Language Model) multi-agent framework. It is an open-source community initiative focused on exploring the scaling laws of agents, enabling the development and interaction of multiple AI agents in a collaborative environment. This framework allows researchers and developers to experiment with various configurations and interactions among agents, facilitating advancements in AI capabilities and understanding.\n"
+          ]
+        }
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "source": [],
+      "metadata": {
+        "id": "5cuumB_bo46c"
+      }
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "## Core Components"
+      ],
+      "metadata": {
+        "id": "biR7z_UuVIpb"
+      }
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "### MemoryRecord"
+      ],
+      "metadata": {
+        "id": "yhZN9zrGVMGw"
+      }
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "The basic data unit in the `CAMEL` memory system.\n",
+        "\n",
+        "Attributes:\n",
+        "\n",
+        "*   `message`: The main content of the record (`BaseMessage`)\n",
+        "*   `role_at_backend`: The role this message played at the OpenAI backend (`OpenAIBackendRole`)\n",
```

Comment:
Perhaps the cookbook just needs some direct usage examples, and more details can be put in the documentation. I think we can leave a document link here.